Changes to use of constant namedb and seq files

RamRS opened the pull request:
These are changes that override the changes from the previous commit/pull-request. Changes have been made in the way objects are pickled and unpickled, and also in the use of CMD line args.
## 

Ram

RamRS committed:
Modified namedb.py and annotate-seqs.py to work with files supplied via command line to annotate-seqs.py

mouse.namedb, mouse.namedb.fullname and mouse.protein.faa were hardcoded into namedb.py.
Now, it picks up from sys.argv[4] and sys.argv[5], where sys.argv is supplied to
annotate-seqs.py

Possible regression bugs if namedb.py is used by scripts other than annotate-seqs.py
that use a different command line argument pattern

RamRS committed:
Modified annotate-seqs, make-namedb and namedb to work on custom seq
file and constant namedb files

make-namedb changed to pickle into constant names.db and fullnames.db
files, and pickle sequence filename into names.db. namedb.py modified to
unpickle correspondingly. Lines expecting extra CMD line args in
annotate-seqs removed.

ctb commented on make-namedb.py:
```diff
@@ -17,7 +18,7 @@
 
 fp = open(outfile, 'w')
-dump(d, fp)
+dump(dict(seqs=sys.argv[1],names=d), fp)
```
Two comments --

since sys.argv[1] is now used twice in the script (and in any case I wrote it badly in the first place), please set a variable to the value at the top, and then reuse the variable down here.  I guess something like 'seqfile = sys.argv[1]'.

second, you don't need to create a dict and then populate it and then dump it.  Just do

dump(seqfile, fp)
dump(names, fp)

following which you can do

seqfile = load(fp)
names = load(fp)

This also eliminates the use of 'get' below...

ctb commented on namedb.py:
```diff
@@ -1,6 +1,8 @@
-mouse_seqs = screed.ScreedDB('mouse.protein.faa')
+data = cPickle.load('names.db')
+
+data_names = data.get('names',[])
```
Here, you should have used 

data_names = data['names']

since 'names' is required and a KeyError should be raised if it's not there.  However, this section of the code should in any case be replaced by sequential load cals as on the line comment above.

RamRS committed:
Pickling no longer uses dict. var seqFile stores seq file name

dict() used in previous commit has been removed, sequential calls to
dump() and load() are used instead. A variable 'seqFile' id used to
store sequence file name from CMD line args.
